stl/src/vector_algorithms.cpp

```diff
@@ -451,6 +464,526 @@ __declspec(noalias) void __cdecl __std_reverse_copy_trivially_copyable_8(
+    }
+
+    static int64_t _Get_any(__m128i _Cur) {
+        return static_cast<uint64_t>(_mm_cvtsi128_si64(_Cur));
+    }
+
+    static uint64_t _Get_v_pos(__m128i _Idx, unsigned long _H_pos) {
+        uint64_t _Array[2];
+        _mm_storeu_si128(reinterpret_cast<__m128i*>(&_Array), _Idx);
+        return _Array[_H_pos >> 3];
+    }
+
+    static __m128i _Cmp_eq(__m128i _First, __m128i _Second) {
+        return _mm_cmpeq_epi64(_First, _Second);
+    }
+
+    static __m128i _Cmp_gt(__m128i _First, __m128i _Second) {
+        return _mm_cmpgt_epi64(_First, _Second);
+    }
+
+    static __m128i _Cmp_lt(__m128i _First, __m128i _Second) {
+        __m128i _Gt = _mm_cmpgt_epi64(_Second, _First); // less or equal
+        __m128i _Eq = _mm_cmpeq_epi64(_First, _Second);
+        return _mm_andnot_si128(_Eq, _Gt);
+    }
+
+    static __m128i _Min(__m128i _First, __m128i _Second, __m128i _Mask) {
+        return _mm_blendv_epi8(_First, _Second, _Mask);
+    }
+
+    static __m128i _Max(__m128i _First, __m128i _Second, __m128i _Mask) {
+        return _mm_blendv_epi8(_First, _Second, _Mask);
+    }
+};
+
+template <_Min_max_mode _Mode, class _Traits>
+auto _Minmax_element(const void* _First, const void* _Last, bool _Sign) noexcept {
+    _Min_max_t _Res                          = {_First, _First};
+    auto _Base                               = static_cast<const _Traits::_Unsigned_t*>(_First);
+    typename _Traits::_Signed_t _Cur_min_val = _Traits::_Init_min_val;
+    typename _Traits::_Signed_t _Cur_max_val = _Traits::_Init_max_val;
+
+    if (_Byte_length(_First, _Last) >= 16 && _Use_sse42()) {
+        const void* _Stop_at = _First;
+
+        size_t _Sse_size     = _Byte_length(_First, _Last) & ~size_t{0xF};
+        size_t _Portion_size = _Sse_size;
+
+        if constexpr (_Traits::_Has_portion_max) {
+            constexpr size_t _Max_portion_size = _Traits::_Portion_max * 16;
+            if (_Portion_size > _Max_portion_size) {
+                _Portion_size = _Max_portion_size;
+            }
+        }
+
+        _Sse_size -= _Portion_size;
+
+        _Advance_bytes(_Stop_at, _Portion_size);
+
+        const __m128i _Inc    = _Traits::_Inc();
+        __m128i _Cur_vals     = _Traits::_Sign_cor(_mm_loadu_si128(reinterpret_cast<const __m128i*>(_First)), _Sign);
+        __m128i _Cur_vals_min = _Cur_vals;
+        __m128i _Cur_idx_min  = _mm_setzero_si128();
+        __m128i _Cur_vals_max = _Cur_vals;
+        __m128i _Cur_idx_max  = _mm_setzero_si128();
+        __m128i _Cur_idx      = _mm_setzero_si128();
+
+        for (;;) {
+            _Advance_bytes(_First, 16);
+            _Cur_idx = _mm_add_epi64(_Cur_idx, _Inc);
+
+            if (_First == _Stop_at) {
+
+                if constexpr (_Mode != _Min_max_mode::_Max_only) {
+                    __m128i _H_min                         = _Traits::_H_min(_Cur_vals_min);
+                    typename _Traits::_Signed_t _H_min_val = _Traits::_Get_any(_H_min);

```

Comment:
Could we get away with `auto`

Reply:
Need to be careful to make all `_Get_any` return Signed type, but yes